src/compiler/checker.ts

```diff
@@ -6454,7 +6454,10 @@ namespace ts {
                     }
                     for (let i = numTypeArguments; i < numTypeParameters; i++) {
                         const mapper = createTypeMapper(typeParameters, typeArguments);
-                        const defaultType = getDefaultFromTypeParameter(typeParameters[i]);
+                        let defaultType = getDefaultFromTypeParameter(typeParameters[i]);
+                        if (defaultType && isTypeIdenticalTo(defaultType, emptyObjectType) && isJavaScriptImplicitAny) {

```

Comment:
would `emptyTypeLiteralType` equality check do the trick?

Reply:
Any `{}` which has an alias symbol (like in `react`) would defeat that. IE,
```
type InternalMembers = {}

export class ComponentBase<P = InternalMembers> {}
```